semgrep-core/src/core/ast/Visitor_AST.ml

```diff
@@ -1003,18 +1002,6 @@ let (mk_visitor : visitor_in -> visitor_out) =
     let v_vinit = v_option v_expr v_vinit in
     let v_vtype = v_option v_type_ v_vtype in
     ()
-  and v_vardef_as_assign_expr ventity = function
-    | VarDef ({ vinit = Some _; _ } as vdef) ->
-        (* A VarDef is implicitly a declaration followed by an assignment expression,
-         * so we should visit the assignment expression as well.
-         *
-         * Note that we cannot treat this as a simple equivalence later, as
```

Comment:
hmm how did you solve this point?

Reply:
If I understood this note correctly, it is saying that expression patterns are only tried against expressions, and statement patterns are only tried against statements, and that's why we can't use an equivalence to match `$X = $E` against `var x = 1;`. But that can be solved by matching both `$X = $E` and also `ExprStmt ($X = $E)`. That is, if I understood the note correctly!